code/game/objects/structures/mirror.dm

```diff
@@ -143,129 +163,141 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/mirror, 28)
+					stylist.body_type = MALE
+					if(stylist.gender != PLURAL)
+						stylist.gender = MALE //we'll update their gender to match their body type- if they want their gender to not match their body type, they can change that using the gender-changing function of the mirror
 				else
 					return
 			H.dna.update_ui_block(DNA_GENDER_BLOCK)
 			H.update_body()
 			H.update_mutations_overlay() //(hulk male/female)
-
-		if("hair")
-			var/hairchoice = tgui_alert(H, "Hairstyle or hair color?", "Change Hair", list("Style", "Color"))
-			if(!user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
+			curse(stylist)
+
+		if("hairstyle")
+			..()
+		
+		if("hair color")
+			var/new_hair_color = input(stylist, "What is the color of our hair again?", "Hair Color",H.hair_color) as color|null
+			if(!stylist.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
 				return
-			if(hairchoice == "Style") //So you just want to use a mirror then?
-				..()
-			else
-				var/new_hair_color = input(H, "Choose your hair color", "Hair Color",H.hair_color) as color|null
-				if(!user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
+			
+
+			if(new_hair_color)
+				stylist.hair_color = sanitize_hexcolor(new_hair_color)
+				stylist.dna.update_ui_block(DNA_HAIR_COLOR_BLOCK)
+				stylist.update_hair()
+				if(curse(stylist))
 					return
-				if(new_hair_color)
-					H.hair_color = sanitize_hexcolor(new_hair_color)
-					H.dna.update_ui_block(DNA_HAIR_COLOR_BLOCK)
-				if(H.gender == "male")
-					var/new_face_color = input(H, "Choose your facial hair color", "Hair Color",H.facial_hair_color) as color|null
-					if(new_face_color)
-						H.facial_hair_color = sanitize_hexcolor(new_face_color)
-						H.dna.update_ui_block(DNA_FACIAL_HAIR_COLOR_BLOCK)
-				H.update_hair()
-
-		if(BODY_ZONE_PRECISE_EYES)
-			var/new_eye_color = input(H, "Choose your eye color", "Eye Color",H.eye_color) as color|null
-			if(!user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
+
+			var/new_face_color = input(stylist, "What is the color of our facial hair again?", "Hair Color",stylist.facial_hair_color) as color|null
+			
+			if(!stylist.canUseTopic(src, BE_CLOSE, FALSE, NO_TK) || !new_face_color)
 				return
-			if(new_eye_color)
-				H.eye_color = sanitize_hexcolor(new_eye_color)
-				H.dna.update_ui_block(DNA_EYE_COLOR_BLOCK)
-				H.update_body()
-	if(choice)
-		curse(user)
-
-/obj/structure/mirror/magic/proc/curse(mob/living/user)
-	return
+
+			stylist.facial_hair_color = sanitize_hexcolor(new_face_color)
+			stylist.dna.update_ui_block(DNA_FACIAL_HAIR_COLOR_BLOCK)
+			stylist.update_hair()
+			curse(stylist)
+
+
+
+		if("eyes")
+			var/new_eye_color = input(stylist, "What is the color of our eyes again?", "Eye Color",stylist.eye_color) as color|null
+			if(!stylist.canUseTopic(src, BE_CLOSE, FALSE, NO_TK) || !new_eye_color)
+				return
+			stylist.eye_color = sanitize_hexcolor(new_eye_color)
+			stylist.dna.update_ui_block(DNA_EYE_COLOR_BLOCK)
+			stylist.update_body()
+			curse(stylist)
```

Comment:
shouldn't you be checking curse's return here?

Reply:
nope, I only need to check its return if there's stuff after it

Reply:
*nod